### ISSUE TITLE
Read transactions cannot immediately obtain the last committed transaction id?

### ISSUE THREAD

**Issue by wanyvic** [cberner/redb#698]
Hi team, can anyone help me with my question?

I have a program that creates a write transaction to submit data to `redb`, and it also provides an API service that creates read transactions to read data when responding to HTTP requests. I found that when my program was committing a write transaction, the latency of my API service increased significantly.

After reading the source code, I discovered a lock conflict that caused this issue.
https://github.com/cberner/redb/blob/a70cfb69f101383932e5876260c2be2489d7f579/src/tree_store/page_store/page_manager.rs#L585

https://github.com/cberner/redb/blob/a70cfb69f101383932e5876260c2be2489d7f579/src/tree_store/page_store/page_manager.rs#L770

During the commit process of a write transaction, `non_durable_commit` locks the `state`, causing a new read transaction created during this period to be unable to obtain `last_committed_transaction_id`.

I'm not sure if this is a performance issue, and how to optimize my service (preferably without reusing read transactions).

**Comment by cberner** [comment]
Can you try this branch and let me know if it improves performance for you? https://github.com/cberner/redb/pull/699

**Comment by wanyvic** [comment]
> Can you try this branch and let me know if it improves performance for you? #699

I tried the code in this branch, but the performance did not improve.
I made a mistake that my code did not reach the `non_durable_commit` function as expected, but instead went to `commit_inner` because I set the write transaction durability level to `Immediate`. This also caused an issue with the `state` lock.

https://github.com/cberner/redb/blob/a70cfb69f101383932e5876260c2be2489d7f579/src/tree_store/page_store/page_manager.rs#L523
